hibernate-core/src/main/java/org/hibernate/loader/ast/internal/MultiIdLoaderStandard.java

```diff
@@ -91,6 +95,20 @@ public EntityMappingType getLoadable() {
 		}
 	}
 
+	private List<T> performSingleLoad(Object id, MultiIdLoadOptions loadOptions, SharedSessionContractImplementor session) {
```

Comment:
EntityPersister already caches the single-id loader.  Is there a reason to not just simply re-use it?

```
entityDescriptor.getIdLoader()...
```

Yes, at the moment it does not expose its `AbstractEntityPersister#singleIdEntityLoader`, but that was initially true for the other loaders which we exposed as needed.

You could even just call `EntityPersister#load` I think